### PR TITLE
Explain how to run conformance tests

### PR DESCRIPTION
**Pull request by wmdietl**
@netdpb Are these changes correct?
Should this also mention `jspecifySamplesTest`?

In our last meeting you had mentioned that all tests should currently pass.
`test` trivially passes, but both `jspecifySamplesTest` and `conformanceTest` fail for me.
I made sure the `samples-google-prototype` branch of `jspecify/jspecify` is checked out and also tried in a freshly checked-out copy of everything.

Do I need to check special branches for other projects?

I also find this [line](https://github.com/jspecify/nullness-checker-for-checker-framework/blob/a09dfe4307acdbf4fde0268c98ee8aa6576911aa/.github/workflows/build.yml#L28) odd, because the instructions say branch `samples-google-prototype` branch of `jspecify/jspecify` needs to be checked out, but this uses `main`.
Also, in the CI output it doesn't really look like the tests are running, does it?

### CODE REVIEW

**Comment by netdpb** [comment]
It looks like those testing instructions are incomplete, and don't run the conformance test. Let me try to edit to make this more accurate.

File: README.md [hunk 1]
```diff
@@ -86,9 +86,18 @@ Note that the `demo` script is not complicated, and illustrates how you can enab
...
 cd $root_dir/nullness-checker-for-checker-framework
 ./gradlew test
 ```
 
+To run the conformance test suite, run:
+
+```sh
+cd $root_dir/nullness-checker-for-checker-framework
+./gradlew conformanceTest
+```
```
Comment (netdpb) [inline README.md:101]:
```suggestion
To run the (incomplete) conformance test suite, check out the `main` branch of `jspecify` and run `conformanceTest`:

```sh
git -C $root_dir/jspecify checkout main
cd $root_dir/nullness-checker-for-checker-framework
./gradlew conformanceTest
```

And then also add:

```
To run the (legacy) "samples" test suite, check out the `samples-google-prototype` branch of `jspecify` and run `jspecifySamplesTest`:

```sh
git -C $root_dir/jspecify checkout samples-google-prototype
cd $root_dir/nullness-checker-for-checker-framework
./gradlew jspecifySamplesTest
```
```

Comment (netdpb) [reply README.md:101]: (I tried to create a PR onto your branch, but I couldn't get it to work.)

File: README.md [hunk 2]
```diff
@@ -86,9 +86,18 @@ Note that the `demo` script is not complicated, and illustrates how you can enab
...
+
 Note that this only passes because the steps above caused you to check out the `samples-google-prototype` branch of `jspecify`. This way you are running only the tests that are actually *expected* to pass.
```
Comment (netdpb) [inline README.md:103]: Oh, and delete this.

**Comment by wmdietl** [comment]
Thanks, these instructions now work for me and both test suites pass.
There is not a lot of output to tell how many tests were actually executed, so I'm not sure anything actually happens.
But the gradle tasks are successful now.

**Comment by wmdietl** [comment]
@netdpb said: "(I tried to create a PR onto your branch, but I couldn't get it to work.)"
In the top-right column it says: "Allow edits and access to secrets by maintainers".
So maybe @cpovirk needs to make you a maintainer?

**Comment by cpovirk** [comment]
_poof_, you're a maintainer, @netdpb

**Review by netdpb — APPROVED** [review]

**Pull request merged**